File: unit-2/sprint-1/day-6-Event Listener & Saving UI State/we/Screen Rotation/app/src/main/java/com/example/screenrotationtest/MainActivity.java
package com.example.screenrotationtest;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    private TextView mTvSaveState;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mTvSaveState = findViewById(R.id.tvSaveState);
        mTvSaveState.setText("Welcome to Masai School");
        Log.d("TAG","onCreate");
    }

    @Override
    protected void onRestoreInstanceState(@NonNull Bundle savedInstanceState) {
        super.onRestoreInstanceState(savedInstanceState);
        Log.d("TAG","onRestore");
        if (savedInstanceState != null){
            String data = savedInstanceState.getString("data");
            mTvSaveState.setText(data);
        }
    }

    @Override
    protected void onSaveInstanceState(@NonNull Bundle outState) {
        super.onSaveInstanceState(outState);
        outState.putString("data","Screen Rotated");
        Log.d("TAG","onSavedInstance");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        Log.d("TAG","onDestroy");
    }
}

File: unit-4/sprint-2/Coroutine/you/Api Call/app/src/main/java/com/example/myapplication/MainActivity.kt
package com.example.myapplication

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.coroutines.*
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class MainActivity : AppCompatActivity() {
    private lateinit var contentList: List<ResponseModelItem>
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        CoroutineScope(Dispatchers.IO).launch {

            val res: String = async {
                var data = getData()
                return@async "$data"
            }.await()

            CoroutineScope(Dispatchers.Main).launch {
                Log.d("onCreate", "onCreate: $res")
            }
        }
    }
    fun getData(): String {
        val apiService = Network.getInstance().create(ApiService::class.java)
        apiService.getUserData().enqueue(object : Callback<ResponseModel> {
            override fun onResponse(
                call: Call<ResponseModel>,
                response: Response<ResponseModel>
            ) {
                response.body()?.run {
                    contentList = response.body()!!
                    tvContent.text = response.body()?.get(0)?.name.toString()
                }
            }
            override fun onFailure(call: Call<ResponseModel>, t: Throwable) {

            }
        })
       return ""
    }
}